x/stake/pool.go

```diff
@@ -4,10 +4,61 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
+// Pool - dynamic parameters of the current state
+type Pool struct {
+	TotalSupply       int64   `json:"total_supply"`        // total supply of all tokens
+	UnbondedShares    sdk.Rat `json:"unbonded_shares"`     // sum of all shares distributed for the Unbonded Pool
+	UnbondingShares   sdk.Rat `json:"unbonding_shares"`    // shares moving from Bonded to Unbonded Pool
+	BondedShares      sdk.Rat `json:"bonded_shares"`       // sum of all shares distributed for the Bonded Pool
+	UnbondedTokens    int64   `json:"unbonded_pool"`       // reserve of unbonded tokens held with validators
+	UnbondingTokens   int64   `json:"unbonding_pool"`      // tokens moving from bonded to unbonded pool
+	BondedTokens      int64   `json:"bonded_pool"`         // reserve of bonded tokens
+	InflationLastTime int64   `json:"inflation_last_time"` // block which the last inflation was processed // TODO make time
+	Inflation         sdk.Rat `json:"inflation"`           // current annual inflation rate
+
+	DateLastCommissionReset int64 `json:"date_last_commission_reset"` // unix timestamp for last commission accounting reset (daily)
+
+	// Fee Related
+	PrevBondedShares sdk.Rat `json:"prev_bonded_shares"` // last recorded bonded shares - for fee calcualtions
+}
+
+func (p Pool) equal(p2 Pool) bool {
+	return p.TotalSupply == p2.TotalSupply &&
```

Comment:
can't we just marshal them both and compare that ?

Reply:
yes, I think it's a bit nicer to do it that way, but not much nicer because we need to define an amino codec because of rational library :( - implemented